### ISSUE TITLE
Question about weight initialization

### ISSUE THREAD

**Issue by raven38** [KAIR-BAIR/dycheck#3]
Hi, thank you for your insightful work and for sharing a reproducible repository!

In the MLP for deformation field,
https://github.com/KAIR-BAIR/dycheck/blob/main/dycheck/models/warping/dense.py#L45
it initializes the weight of the last layer by `jax.nn.initializers.uniform(scale=1e-4)`, meaning $\mathcal{U}(0, 10^{-4})$
However, the Nerfies paper says "We therefore initialize the
weights of the last layer of the MLP from $\mathcal{U}(−10^{−5}, 10^{−5})$
to initialize the deformation near the identity." in the last paragraph of Section 3.2.
https://arxiv.org/abs/2011.12948

I think the implementation is different from the paper.
Initializing weights with only positive values leads to a far away from the identity. 

Why is the deformation field initialized by $\mathcal{U}(0, 10^{-4})$?
Also, does the initialization work well?

Thanks.

**Comment by hangg7** [comment]
Hi Kai, thanks for your question and email (sorry that I missed your issue before). Below is the original response email for other folks interested in this issue:

### Q1: U(0, 1e-4) instead of U(-1e-5, 1e-5)?
Nice catch! The latter will indeed be better since it normalized the initial prediction and thus has (potentially) less bias. In our implementation, we follow exactly what Nerfies and HyperNeRF did in their code instead of the paper. Please have a look at [here](https://github.com/google/hypernerf/blob/main/hypernerf/warping.py#L154-L155).

### Q2: Does the initialization work well?
We are able to reproduce the original results from those paper using our code. But yes, both results would have this slight less-ideal initialization. My guess is that because this bias is so small (1e-4), it will not affect results much. But I am curious in what you find.

Best!
Hang
